rundeckapp/grails-app/services/rundeck/services/ExecutionService.groovy

```diff
@@ -1248,6 +1258,35 @@ class ExecutionService implements ApplicationContextAware, StepExecutor, NodeSte
         }
     }
 
+    private String getOutputLimit(boolean logGlobalCongif, ScheduledExecution scheduledExecution, boolean limitFramework) {
+        String outputLimit = ""
+        if (logGlobalCongif) {
+            outputLimit = frameworkService.getProjectLogOutputLimit()
+            limitFramework = true
+        } else {
+            outputLimit = scheduledExecution?.logOutputThreshold
+            if (!outputLimit) {
+                outputLimit = frameworkService.getProjectLogOutputLimit()
+                limitFramework = true
+            }
+        }
+        log.info("Logging output limit: ${outputLimit}")
+        return outputLimit
+    }
+
+    private String getLimitAction(boolean logGlobalConfig, boolean limitFramework, ScheduledExecution scheduledExecution) {
+        String limitAction = scheduledExecution?.logOutputThresholdAction
+        if (logGlobalConfig) {
+            limitAction = frameworkService.getProjectLogLimitAction()
+        } else {
+            if (limitFramework) {
+                limitAction = frameworkService.getProjectLogLimitAction() ?: scheduledExecution?.logOutputThresholdAction
+            }
+        }
+        log.info("Logging limit action: ${limitAction}")
+        return limitAction ?: LoggingThreshold.ACTION_HALT
+    }
```

Comment:
The logic here should also be changed.

1. if a global limit is set, it should default to action Truncate
2. if a job limit is set, it should use the job action, but a global action of Halt should take precedence

i.e.
job limit & no global limit => job action
job limit & global limit & global action halt => halt
job limit & global limit & global action truncate => job action
job limit & global limit & no global action =>  job action

Reply:
Done... added an extra validation:
no job limit & global limit => global limit action (default: truncate)